docs/dsl/example.md

```diff
@@ -0,0 +1,438 @@
+| - | - |
+| 1234 | pf |
+| 3456 | az |
+| 4567 | az |
+| 5678 | pf |
+
+```
+cohort.first_date = first_covid_imm.select_column(imms.date)  # PatientSeries
+```
+
+| patient | cohort.first_date |
+| - | - |
+| 1234 | 2021-03-19 |
+| 3456 | 2021-04-01 |
+| 4567 | 2021-06-23 |
+| 5678 | 2021-07-01 |
+
+```
+cohort.first_date + 28  # PatientSeries
+```
+
+| patient | cohort.first_date + 28|
+| - | - |
+| 1234 | 2021-04-16 |
+| 3456 | 2021-04-29 |
+| 4567 | 2021-07-21 |
+| 5678 | 2021-07-29 |
+
+```
+imms.date > (cohort.first_date + 28)  # PatientSeries
+```
+
+| id | date > cohort.first_date + 28 |
+| - | - |
+| 43 | F |
+| 44 | T |
+| 45 | F |
+| 46 | T |
+| 47 | T |
+| 48 | - |
+| 49 | - |
+| 50 | T |
+| 51 | T |
+| 52 | T |
+| 53 | - |
+| 54 | F |
+| 55 | T |
+| 56 | F |
+| 57 | T |
+| 58 | F |
+
+To see where this came from, consider this table, and note that the values of the `cohort.first_date + 28` column come from the values corresponding to each row's patient.
+
+| id | patient | code | date | cohort.first_date + 28 | date > cohort.first_date + 28 |
+| - | - | - | - | - | - |
+| 43 | 1234 | pf | 2021-03-19 | 2021-04-16 | F |
+| 44 | 4567 | pox | 2021-10-10 | 2021-07-21 | T |
+| 45 | 4567 | az | 2021-06-23 | 2021-07-21 | F |
+| 46 | 1234 | flu | 2021-10-15 | 2021-04-16 | T |
+| 47 | 1234 | pf | 2021-07-04 | 2021-04-16 | T |
+| 48 | 2345 | flu | 2021-03-25 | - | - |
+| 49 | 2345 | flu | 2021-08-19 | - | - |
+| 50 | 5678 | pox | 2021-10-11 | 2021-07-29 | T |
+| 51 | 4567 | flu | 2021-12-14 | 2021-07-21 | T |
+| 52 | 3456 | pf | 2021-09-09 | 2021-04-29 | T |
+| 53 | 2345 | flu | 2021-11-02 | - | - |
+| 54 | 5678 | pf | 2021-07-01 | 2021-07-29 | F |
+| 55 | 3456 | pox | 2021-11-03 | 2021-04-29 | T |
+| 56 | 3456 | az | 2021-04-01 | 2021-04-29 | F |
+| 57 | 5678 | az | 2021-12-30 | 2021-07-29 | T |
+| 58 | 5678 | pf | 2021-07-02 | 2021-07-29 | F |
+
+```
+covid_imms.filter(imms.date > (cohort.first_date + 28))  # EventFrame
+```
+
+| id | patient | code | date |
+| - | - | - | - |
+| 45 | 4567 | az | 2021-06-23 |
+| 47 | 1234 | pf | 2021-07-04 |
+| 52 | 3456 | pf | 2021-09-09 |
+| 57 | 5678 | az | 2021-12-30 |
+
+```
+second_covid_imm = (
+    covid_imms.filter(imms.date > (cohort.first_date + 28))
+    .sort_by(imms.date)
+    .first_for_patient()
+)  # AggregatedEventFrame
+```
+
+| id | patient | code | date |
+| - | - | - | - |
+| 47 | 1234 | pf | 2021-07-04 |
+| 52 | 3456 | pf | 2021-09-09 |
+| 57 | 5678 | az | 2021-12-30 |
+
+```
+cohort.second_date = second_covid_imm.select_column("date")  # PatientSeries
+```
+
+| patient | cohort.second_date |
+| - | - |
+| 1234 | 2021-07-04 |
+| 3456 | 2021-09-09 |
+| 5678 | 2021-12-30 |
+
+```
+cohort.second_date = second_covid_imm.select_column("code")  # PatientSeries
+```
+
+| patient | cohort.second_code |
+| - | - |
+| 1234 | pf |
+| 3456 | pf |
+| 5678 | az |
+
+```
+cohort.codes_match = categorise(
+    {
+        "yes": cohort.first_code == cohort.second_code,
+        "no": True,
+    }
```

Comment:
question: Does `categorise()` work by finding the first `True` value and assigning the key?